src/main/flight/alt_hold.c

```diff
@@ -22,118 +22,50 @@
-    const float pOut = pidCoefs->Kp * altitudeErrorCm;
-
-    // I
-    // input limit iTerm so that it doesn't grow fast with large errors
-    // very important at the start if there are massive initial errors to prevent iTerm windup
-
-    // much less iTerm change for errors greater than 2m, otherwise it winds up badly
-    const float itermNormalRange = 200.0f; // 2m
-    const float itermRelax = (fabsf(altitudeErrorCm) < itermNormalRange) ? 1.0f : 0.1f;
-    pidIntegral += altitudeErrorCm * pidCoefs->Ki * itermRelax * taskIntervalSeconds;
-    // arbitrary limit on iTerm, same as for gps_rescue, +/-20% of full throttle range
-    // ** might not be needed with input limiting **
-    pidIntegral = constrainf(pidIntegral, -200.0f, 200.0f); 
-    const float iOut = pidIntegral;
-
-    // D
     // boost D by 'increasing apparent velocity' when vertical velocity exceeds 5 m/s ( D of 75 on defaults)
     // usually we don't see fast ascend/descend rates if the altitude hold starts under stable conditions
     // this is important primarily to arrest pre-existing fast drops or climbs at the start;
-
     float vel = getAltitudeDerivative(); // cm/s altitude derivative is always available
     const float kinkPoint = 500.0f; // velocity at which D should start to increase
     const float kinkPointAdjustment = kinkPoint * 2.0f; // Precompute constant
     const float sign = (vel > 0) ? 1.0f : -1.0f;
     if (fabsf(vel) > kinkPoint) {
         vel = vel * 3.0f - sign * kinkPointAdjustment;
     }
+    const float verticalVelocity = vel;
 
-    const float dOut = pidCoefs->Kd * vel;
+    const float targetAltitudeCm = altHoldState.targetAltitudeCm;
+    const float targetChangeRate = altHoldState.targetAltitudeAdjustRate;
 
-    // F
-    // if error is used, we get a 'free kick' in derivative from changes in the target value
-    // but this is delayed by the smoothing, leading to lag and overshoot.
-    // calculating feedforward separately avoids the filter lag.
-    // Use user's D gain for the feedforward gain factor, works OK with a scaling factor of 0.01
-    // A commanded drop at 100cm/s will return feedforward of the user's D value. or 15 on defaults
-    const float fOut = pidCoefs->Kf * altHoldState.targetAltitudeAdjustRate;
-    
-    // to further improve performance...
-    // adding a high-pass filtered amount of FF would give a boost when altitude changes were requested
-    // this would offset motor lag and kick off some initial vertical acceleration.
-    // this would be exactly what an experienced pilot would do.
-
-    const float output = pOut + iOut - dOut + fOut;
-    DEBUG_SET(DEBUG_ALTHOLD, 4, lrintf(pOut));
-    DEBUG_SET(DEBUG_ALTHOLD, 5, lrintf(iOut));
-    DEBUG_SET(DEBUG_ALTHOLD, 6, lrintf(dOut));
-    DEBUG_SET(DEBUG_ALTHOLD, 7, lrintf(fOut));
-
-    return output; // motor units, eg 100 means 10% of available throttle 
+    altitudeControl(targetAltitudeCm, taskIntervalSeconds, verticalVelocity, targetChangeRate);
```

Comment:
```suggestion
    altitudeControl(altHoldState.targetAltitudeCm, taskIntervalSeconds, vel, altHoldState.targetAltitudeAdjustRate);
```

Reply:
Yes, simpler, wasn't sure why I put them into the floats. Thanks.